packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -451,15 +450,11 @@ export class NxMonorepoProject extends TypeScriptProject {
           default: ["{projectRoot}/**/*"],
           ...options.nxConfig?.namedInputs,
         },
-        targetDefaults: options.nxConfig?.targetDefaults,
-        targetDependencies: {
-          build: [
-            {
-              target: "build",
-              projects: "dependencies",
-            },
-          ],
-          ...(this.nxConfig?.targetDependencies || {}),
+        targetDefaults: {
+          build: {
+            dependsOn: ["^build"],
+          },
+          ...(this.nxConfig?.targetDefaults || {}),
```

Comment:
FYI: `|| {}` is redundant for object spread, only necessary for array spread.

Reply:
Nice.